spec/p2p/reactor-survey.md

```diff
@@ -0,0 +1,247 @@
+
+
+#### [CM-PROTOCOL-MR-COMM.0]
+
+Gossiping is done by remembering for each peer which was the last element in the local list of transaction that was sent to a peer (there are condition where the pointer to the last element is reset). We then send transactions after this last element to the peer. 
+
+> TODO: seems best effort send, i.e., no ACKs, no reliable communication (e.g., full messages queues). We should clarify.
+
+### Requirements on the p2p layer
+
+Similar to the consensus reactor, the discussion above entails  **local requirements** that express needs as connections to neighbors.
+
+#### [CM-REQ-CR+P2P-STABLE.0]
+In order to make progress in sending the list of transactions to a peer, the p2p layer must ensure that we stay connected to a peer sufficiently long.
+
+Similar to the consensus reactor, [CM-REQ-MEMP-GOSSIP.0] translates into global connectivity requirements. However, [CM-REQ-CR+P2P-STABILITY.0] postulates a stable connected overlay for a continuous interval of time (to ensure time-bounded communication in good periods). A priori, for the transactions, we just require that over time all transactions reach the validators (reliable communication; not-necessarily time bounded), which does not imply connectivity within such an interval.
+
+> However, expectations from the users might be different. With MEV etc., we will have time expectations regarding transactions processing. This will impose stronger constraints again.
+
+## Evidence reactor
+
+All that we found for the mempool also applies for the evidence reactor. 
+Evidence are specific transactions that are used by the application (e.g., the staking module) to punish misbehavior. Typical applications (CosmosSDK chains) thus would benefit from timeliness: due to proof-of-stake and unbonding periods, a specific evidence transaction has an expiration date after which the application cannot act upon it.
+
+There are several crucial things that are outside of the control of the evidence reactor
+1. (application level) unbonding period
+1. BFT time (written into the blocks) is under control of consensus
+1. when evidence is put into a block (similar to the mempool we may assume that correct proposers always put all the evidence they are aware of in a block, but CometBFT cannot enforce that)
+1. when evidence is submitted
+
+In face of this uncertainty, there are only few reasonable requirement we can postulate
+
+#### [CM-PROTOCOL-ER-COMM.0]
+1. An evidence transaction reaches all validator nodes as "fast as possible".
+1. Evidence is not written into a block after it has expired
+
+> With more control over the four points, in principle we might define a real-time property which formalizes "if evidence is submitted at least a day before it expires, then evidence is written into a block before it expires". 
```

Comment:
Nothing guarantees that evidence submitted at some wall-clock time X will be written into a block before some later wall-clock time Y.

Reply:
I agree. The statement also doesn't contradict this.

Reply:
Does the statement not assert that evidence submitted at wall-clock time X (where X is at least a day before the expiration time of that evidence) will be written into a block before some later wall-clock time Y (where Y is the expiration time of the evidence)"?